Source/cielim-python/python-driver/proto_interface.py

```diff
@@ -0,0 +1,171 @@
+        target_velocity = np.zeros(3)
+        spacecraft_position = np.array(self.cielim_msg.spacecraft.position)
+        spacecraft_velocity = np.array(self.cielim_msg.spacecraft.velocity)
+        camera_position = np.array(self.cielim_msg.camera.cameraPositionInBody)
+        camera_orientation = np.array(self.cielim_msg.camera.bodyFrameToCameraMrp)
+        for body in self.cielim_msg.celestialBodies:
+            if body.bodyName.lower() == target_name.lower():
+                target_position = np.array(body.position)
+                target_velocity = np.array(body.velocity)
+
+        primary = target_position - (spacecraft_position - camera_position)
+        primary /= np.linalg.norm(primary)
+
+        secondary = target_velocity - spacecraft_velocity
+        secondary /= np.linalg.norm(secondary)
+
+        BN = body_to_inertial_for_pointing(primary, secondary, mrp_to_dcm(camera_orientation))
+        self.cielim_msg.spacecraft.ClearField("attitude")
+        [self.cielim_msg.spacecraft.attitude.append(item) for item in dcm_to_mrp(BN)]
+
+    def propagate(self, end_time, gravitational_parameter=0):
+        """
+        Function to propagate the camera position.
+        If the target is not found among the bodies, the pointing will point to the zero inertial point
+        :param end_time: time (seconds) to go to
+        :param gravitational_parameter: optional, reset gravitational parameter (SI)
+        :param target_name: optional, reset target (string)
+        :return:
+        """
+        if gravitational_parameter != 0:
+            self.gravitational_parameter = gravitational_parameter
+        elif self.gravitational_parameter == "":
+            print("No gravitational parameter provided, rectilinear trajectory assumed")
+
+        initial_state = np.array(list(self.cielim_msg.spacecraft.position) + list(self.cielim_msg.spacecraft.velocity))
+        final_state = propagate_cartesian(self.gravitational_parameter, initial_state, 0, end_time)
+        self.cielim_msg.spacecraft.ClearField("position")
+        self.cielim_msg.spacecraft.ClearField("velocity")
+        [self.cielim_msg.spacecraft.position.append(item) for item in final_state[:3]]
+        [self.cielim_msg.spacecraft.velocity.append(item) for item in final_state[3:]]
+
+    def propagate_and_stare(self, end_time, gravitational_parameter=0, target_name=""):
+        """
+        Function to propagate the camera position and maintain the pointing of the camera to the target in the scene. 
+        If the target is not found among the bodies, the pointing will point to the zero inertial point
+        :param end_time: time (seconds) to go to
+        :param gravitational_parameter: optional, reset gravitational parameter (SI)
+        :param target_name: optional, reset target (string)
+        :return: 
+        """
+        self.propagate(end_time, gravitational_parameter)
+        self.look_at_target(target_name)
+
+    def return_message(self):
```

Comment:
:large_orange_diamond: I think this should be something like `get_current_scene_message`. Avoiding `return` as it's already a key word invoking ideas of function closure.